cmds/switch_root/switch_root.go

```diff
@@ -23,140 +26,171 @@ func usage() string {
 	return "switch_root [-h] [-V]\nswitch_root newroot init"
 }
 
-// littleDoctor recursively deletes everything at "path" that
-// is in the same file system as "fs".
-func littleDoctor(path string, fsType int64) error {
-	var pathFS syscall.Statfs_t
+// get device for file descriptor
+func getDev(fd int) (dev uint64, err error) {
+	var stat syscall.Stat_t
 
-	if err := syscall.Statfs(path, &pathFS); err != nil {
-		return err
+	if err := syscall.Fstat(fd, &stat); err != nil {
+		return 0, err
 	}
 
-	if int64(pathFS.Type) != fsType {
-		return nil
+	return stat.Dev, nil
+}
+
+// recursively deletes a directory and everything in it
```

Comment:
super nit: Go style comments are usually full sentences with proper capitalization and full stops. Sentence fragments are undesirable. This is especially because comments are often exported as godocs (see e.g. https://godoc.org/github.com/u-root/u-root/pkg/pxe). This applies to all other comments in a Go file just for consistency.

also, function comments commonly start with the identifier of the function. e.g.:

```go
// recursiveDelete deletes a directory identified by `fd` and everything in it.
//
// This function only deletes files on the same device as `fd`.
func recursiveDelete(fd int) error
```